indexd/index/blueprint.py

```diff
@@ -248,7 +248,7 @@ def get_index_record(record):
     Returns a record.
     """
 
-    ret = blueprint.index_driver.get(record)
+    ret = blueprint.index_driver.get_with_nonstrict_prefix(record)
```

Comment:
i think we should also call `get_with_nonstrict_prefix` instead of `get` in `get_bundle_record_with_id` and `indexd/blueprint.get_record`. maybe even in `get_dos_record` for consistency, although that one is not used anymore

Reply:
Sorry, just realizing I missed this after the first review. I can make these changes, but I'm not familiar with what those methods do; would this require any additional testing, or should the existing cases cover it?

Reply:
- `indexd.blueprint.get_record`: instead of only testing the `https://.../index/<id>` endpoint, the `https://.../index/index/<id>` should also be tested
- `get_bundle_record_with_id`: yeah testing this one would require creating a bundle first
- `get_dos_record` let's not update it, better not break legacy code

Reply:
@paulineribeyre regarding the first point, wouldn't it be the other way around? `https://.../index/index/<id>` is the one being updated to the use the new method in this PR; `https://.../index/<id>` hasn't been touched, and from the background info I thought we wanted to leave that one as-is.

Reply:
I think I see the confusion, my comments in the PR body are from the perspective of `/` being the indexd root, not the entire deployment.

Reply:
right sorry, the other way around

why from the background info?

Reply:
missing `indexd.blueprint.get_record`

Reply:
done, testing